### ISSUE TITLE
documentation: describe non-python dependencies in installation documentation

### ISSUE THREAD

**Issue by draeath** [snakemake/snakemake#1328]
**Is your feature request related to a problem? Please describe.**
I am a system administrator who is trying to learn / play with snakemake in an effort to develop some cloud-backed automated workflows for our scientists. I do not use conda in my own environment and my usage of other tools (such as [pyenv](https://github.com/pyenv/pyenv)) and the intended execution environment (kubernetes) makes me resistant to touch conda. I need to _understand_ how this tool is installed so that I can properly support it in such an environment rather than just "push button receive snakemake" a-la the conda instructions.

Currently the [documentation page for installation](../blob/main/main/docs/getting_started/installation.rst) merely says the following regarding non-conda installation:

```
Installation via pip
====================

Instead of conda, snakemake can be installed with pip.
However, note that snakemake has non-python dependencies, such that the pip based installation has a limited functionality if those dependencies are not manually installed in addition.
```

This is rather unhelpful as-is.

**Describe the solution you'd like**
It would be very nice if any detail whatsoever on what these dependencies are was added to the documentation.

I realize I could just ask what the dependencies are on StackOverflow, but I feel like this is the sort of thing that should go into the docs so others don't have to go looking - even if this was just to refer to a file here on github with the information. Presumably if one was familiar with conda, they could determine this from whichever file(s) describes this to conda, but I am not one familiar with that.

**Describe alternatives you've considered**
N/A

**Additional context**
I am intending to develop/test a means to automate the elastic expansion/contraction of compute resources via [kubernetes cloud execution](https://snakemake.readthedocs.io/en/stable/executing/cloud.html#generic-cloud-support-via-kubernetes). I have a cluster doing exactly that with normal kubernetes workloads, and was stymied by this documentation shortfall when trying to bring snakemake into the mix.

**Comment by aryarm** [comment]
Hi @draeath ,

Every conda package has a `meta.yaml` _recipe_ containing metadata for building the package along with any dependencies.

The documentation could probably be altered to provide a link to the conda recipe for Snakemake on bioconda:
https://bioconda.github.io/recipes/snakemake/README.html
Does that have the information you're looking for? If not, what about the `meta.yaml` file linked from it?

Regarding whether you should install Snakemake via pip or conda:
There is a second reason to install conda besides the one that you cite. To quote the installation page,
> it also enables Snakemake to handle software dependencies of your workflow

**tldr:** When conda is installed, Snakemake has the ability to run each step of a pipeline within an isolated conda environment, where each environment is generated from a list of packages in a conda `environment.yml` file.

imo, this is one of Snakemake's most powerful features, so I would highly recommend it.

**Comment by draeath** [comment]
@aryarm that link (I actually prefer the yaml but both are great!) is just what I would need! Perfect. I'd be happy with the documentation just mentioning/linking to them in addition to the current verbiage. I could throw a PR together?

----

I'd have to consider if that environment management capability would be useful. So far we create containers ourselves and Snakemake "just" runs them, and I actually enjoy the environment building and execution itself being separated. But my perspective is a little different than your typical user I would bet - I'm already using things like Ansible, Terraform, and Packer to do that sort of thing.
